File: Poseidon_SpringAPI/src/main/java/fr/orionbs/models/Rule.java
package fr.orionbs.models;

import lombok.*;

import javax.persistence.*;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
@Entity(name = "rule")
public class Rule {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    private String name;
    private String description;
    private String json;
    private String template;
    private String sqlStr;
    private String sqlPart;

    @Override
    public String toString() {
        return "Rule{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", description='" + description + '\'' +
                ", json='" + json + '\'' +
                ", template='" + template + '\'' +
                ", sqlStr='" + sqlStr + '\'' +
                ", sqlPart='" + sqlPart + '\'' +
                '}';
    }
}

File: Poseidon-AngularFront/src/app/rule/rule.ts
export class Rule {
	id?: number;
	name?: string;
    description?: string;
    json?: string;
    template?: string;
    sqlStr?: string;
    sqlPart?: string;
	
	constructor(name?: string, description?: string, json?: string, template?: string, sqlStr?: string, sqlPart?: string) {
		this.name = name;
        this.description = description;
        this.json = json;
        this.template = template;
        this.sqlStr = sqlStr;
        this.sqlPart = sqlPart;
	}

}
